### DISCUSSION TITLE
Latest version of prism 8.0.0.1909 exception on contentcontrol region for both framework 4.7 and .Net 5.0 applications?

### DISCUSSION THREAD

**Discussion by stevensrf11** [PrismLibrary/Prism#2382] [WPF]
I created a WPF .Net 5 application. The WPF application is identical to the Sample 2 Region demo which is a .Net Core 3.1 Application. When running the .Net WPF application and exception is thrown saying the region cannot be created for the ContentControl.

I made a similar WPF .Net Framework 4.7 Application using the same version of Prism and the same exception was thrown.

Has anyone else heard reports on this happening from anything ?

Strange thing is if I go Sample2 Region demo and change it to a .Net 5 application is runs ok. The exception does not occur.

And if I create a .Net Core 3.1 Wpf application and chave it to a  .Net 5 application it also runs ok

InnerException {"An exception occurred while creating a region with name 'ContentRegion'. The exception was: System.Collections.Generic.KeyNotFoundException: The IRegionAdapter for the type System.Windows.Controls.StackPanel is not registered in the region adapter mappings. You can register an IRegionAdapter for this control by overriding the ConfigureRegionAdapterMappings method in the bootstrapper.\r\n at Prism.Regions.RegionAdapterMappings.GetMapping(Type controlType) in G:\prism folders\prism8net5\Prism-master\src\Wpf\Prism.Wpf\Regions\RegionAdapterMappings.cs:line 80\r\n at Prism.Regions.Behaviors.DelayedRegionCreationBehavior.CreateRegion(DependencyObject targetElement, String regionName) in G:\prism folders\prism8net5\Prism-master\src\Wpf\Prism.Wpf\Regions\Behaviors\DelayedRegionCreationBehavior.cs:line 132. "} System.Exception {Prism.Regions.Behaviors.RegionCreationException}

**Comment by stevensrf11** [comment]
Well I found the problem for the .Net Framewrok,
If the Project  has a period in it the the exception occurs.
For Example name a WPF Project  Wpf.SampleApp the add prism unity and give a region name to a content control and the exception occurs. That sucks.

**Reply by stevensrf11** [reply]
Well I found the previous email about the fix for .Net Framework may be incorrect.
It  seems if the ContentControl Content is set like below the exception is thrown. Like Shown below
<Window x:Class="WpfApp1.Views.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:prism="http://prismlibrary.com/"
        Title="Shell" Height="350" Width="525">
    <Grid>
        <ContentControl prism:RegionManager.RegionName="ContentRegion" Content="Shell"/>
    </Grid>
</Window>
If the ContentControl Content is not shown at all as shown below the exception does not occurr
<Window x:Class="WpfApp1.Views.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:prism="http://prismlibrary.com/"
        Title="Shell" Height="350" Width="525">
    <Grid>
        <ContentControl prism:RegionManager.RegionName="ContentRegion" />
    </Grid>
</Window>

**Reply by stevensrf11** [reply]
[WpfApp1.zip](https://github.com/PrismLibrary/Prism/files/6296801/WpfApp1.zip)

 Attached is a sample program showing the answer in the Above message the code was suppose to be.
In the previous reply the code with the ContentControl did not get posted.

**Comment by brianlagunas** [comment]
See #2389
